python/src/scipp/coords.py

```diff
@@ -0,0 +1,151 @@
+# SPDX-License-Identifier: BSD-3-Clause
```

Comment:
Should add options to
- remove intermediate results
- not rename dimensions

Can be done in a later step, the current implementation is usable without them.